src/Configuration/Config.Ini/src/IniConfigurationStreamLoader.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Extensions.Configuration.Ini
+{
+    /// <summary>
+    /// An INI file based <see cref="IConfigurationStreamLoader"/>.
+    /// </summary>
+    public class IniConfigurationStreamLoader : IConfigurationStreamLoader
```

Comment:
Should we ship this? (Using an ini file from a stream doesn't seem super useful to me, but then again the refactoring into this doesn't harm, so...)

Reply:
This is indirectly part of https://github.com/aspnet/Extensions/issues/1631 which would improve the Generic Host experience in other app models such as Xamarin where you don't always have a file path for a configuration file. If `.ini` is part of our supported set of config providers, I think it should work with an arbitrary stream too.

Reply:
Fair enough